server/src/main/java/org/opensearch/cluster/metadata/AutoExpandReplicas.java

```diff
@@ -181,6 +181,10 @@ public static Map<Integer, List<String>> getAutoExpandReplicaChanges(Metadata me
         Map<Integer, List<String>> nrReplicasChanged = new HashMap<>();
 
         for (final IndexMetadata indexMetadata : metadata) {
+            // Skip the replica auto-expansion for indices in search_only mode with the SEARCH_ONLY block
+            if (indexMetadata.getSettings().getAsBoolean(IndexMetadata.INDEX_BLOCKS_SEARCH_ONLY_SETTING.getKey(), false)) {
```

Comment:
please put this check inside the existing if block so we aren't checking it for closed indices.  Also, we are now very frequently looking up this setting, can we look it up in IndexMetadata ctor and set a property? Or even just an accessor would go a long way here, like the existing: 

```
    public long getCreationDate() {
        return settings.getAsLong(SETTING_CREATION_DATE, -1L);
    }
```

Reply:
I have moved to the inner `if` block, also yes we can have something like and use it across the code.
```
    public boolean getIndexSearchOnlyBlock() {
        return settings.getAsBoolean(INDEX_BLOCKS_SEARCH_ONLY_SETTING.getKey(), false);
    }
```

Reply:
I would do a follow up PR for this change WDYT @mch2 ?